### PR TITLE
Signin form best practice

### LINKED ISSUE #2886: content: sign-up form best practices [2020-05-29]
* [codelab draft](https://docs.google.com/document/d/1nFIXh8pQFIiR9LpNu5BWeQvNiuy75Afq0VK6cInutx4/edit#heading=h.9yh4lzlezs0q)
* [guide draft](https://docs.google.com/document/d/10pxBi8xJ51uxmQNEK3uFp5G0VogP37z1uYF7lvvAxec)

### PR DESCRIPTION
**Pull request by samdutton**
Fixes #2886

### CODE REVIEW

**Comment by agektmr** [comment]
I started reviewing this but it looks pre-mature. Ping me when you think it's ready to review.

**Comment by samdutton** [comment]
HI @agektmr -- should be OK for review. 

https://deploy-preview-2889--web-dev-staging.netlify.app/signin-form-best-practice

One thing I noticed: screenshot image widths are odd/inconsistent. I'll do a PR to fix them.

**Comment by kaycebasques** [comment]
@agektmr I'm going to propose some significant changes to the codelab so in this case it might be more efficient for you to hold off on your review

**Comment by samdutton** [comment]
> I'm going to propose some significant changes to the codelab

Thanks for looking at this Kayce. Significant changes welcome!

**Comment by kaycebasques** [comment]
List of tasks that we should eventually get to:

- [ ] Use `login` instead of `sign-in` because it appears to be [the more common Search term](https://trends.google.com/trends/explore?geo=US&q=login,signin,sign-in)

**Comment by agektmr** [comment]
Yeah, I vaguely remember Google proactively chose the term "sign-in" over "login" for some reasons. We should keep using "sign-in" unless there's specific reasons.

**Comment by kaycebasques** [comment]
@agektmr it seems like "login" would yield significantly better SEO than "sign-in" based on that Google Trends data.

File: src/site/content/en/blog/signin-form-best-practice/index.md [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+codelabs:
+  - codelab-signin-form
+---
+
+
+If users ever need to log in to your site, then good sign-in form design is critical.
+
+This is especially true for people on poor connections, on mobile, in a hurry, or under stress.
+
+Poorly designed sign-in forms get high bounce rates. Each bounce means a lost and disgruntled user—not just a missed sign-in opportunity.
```
[the remark this answers is not in the document]
Comment (kaycebasques) [reply src/site/content/en/blog/signin-form-best-practice/index.md:28]: @agektmr you should hold off reviewing this until my edits/suggestions from the Google Doc are incorporated into this markdown.

**Comment by kaycebasques** [comment]
@samdutton please re-assign this to me when it's ready for another review

**Comment by samdutton** [comment]
Oops! Sorry. Will undo.

BTW: running npm audit, there seem to be some vulnerabilities. I could do a
separate PR to fix that — or not, if that would be a problem.

**Comment by robdodson** [comment]
no worries. yeah I think doing a separate PR to update dependencies would be nice if you have time :)

**Comment by kaycebasques** [comment]
@samdutton is this ready for another review or are you still working on it?

**Comment by samdutton** [comment]
I'm still working on the codelab, but the article is ready for another
review now.

File: src/site/content/en/blog/sign-in-form-best-practices/index.md [hunk 2]
```diff
@@ -0,0 +1,540 @@
+---
+title: Sign-in form best practices
```
Comment (kaycebasques) [inline src/site/content/en/blog/sign-in-form-best-practices/index.md:2]: @samdutton I changed the URL from `signin-form-best-practice` to `sign-in-form-best-practices`

File: src/site/content/en/blog/sign-in-form-best-practices/index.md [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+This article is about frontend best practice. It does not explain how to build backend services to authenticate users, store their credentials or manage their accounts.
+
+[12 best practices for user account, authorization and password management](https://cloud.google.com/blog/products/gcp/12-best-practices-for-user-account) outlines core principles for running your own backend.
+
+If you have users in different parts of the world, you need to consider localizing your site's use of third party identity services as well as its content.
+{% endAside %}
+
+
+
+## Use meaningful HTML {​: #use-meaningful-html }
```
Comment (kaycebasques) [inline src/site/content/en/blog/sign-in-form-best-practices/index.md:68]: I'm trying to figure out why these section IDs aren't working as expected…

**Comment by kaycebasques** [comment]
@samdutton I rebased this branch. It might be easiest to delete it:

```
git checkout master
git branch -D signin-form-best-practice
```

And then check it out again:

```
git fetch origin
git checkout -b signin-form-best-practice -t origin/signin-form-best-practice
```

**Comment by kaycebasques** [comment]
@samdutton I fixed the custom section IDs (e.g. `## Foo {: #hello }`). My guess is that your `{`, `:`, or `}` character is slightly different.

**Comment by kaycebasques** [comment]
@samdutton I made various minor edits (e.g. breaking lines at 100 characters). It looks like you still have notes to yourself. Please go through the blog post and resolve those and then ping me and I will edit again. We can publish the blog post sooner and the codelab later if you would like to get the post out. We could just mark the codelab as a draft.

**Comment by kaycebasques** [comment]
* The blog post is now good to go.
* I commented out references to the codelab and marked the codelab as a draft.

**Comment by agektmr** [comment]
Sorry about my lagged action, but I'm reviewing now.
I see a message "No project found at signin-form-codelab-5, perhaps a typo?" at https://deploy-preview-2889--web-dev-staging.netlify.app/sign-in-form-best-practices/ is this expected?

File: src/site/content/en/blog/codelab-sign-in-form-best-practices/index.md [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+updated: 2020-06-25
+description: Use cross-platform browser features to build a simple email/password sign-in form that's secure, accessible and easy to use.
+tags:
+  - forms
+  - identity
+  - layout
+  - mobile
+  - security
+  - ux
+glitch: sign-in-form-codelab-0
```
Comment (kaycebasques) [inline src/site/content/en/blog/codelab-sign-in-form-best-practices/index.md:16]:
```suggestion
glitch: sign-in-form-codelab-0
glitch_path: index.html
```

**Comment by kaycebasques** [comment]
We're continuing the good fight over in #3370

**Pull request closed without merging**